src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+
+        #region Fields
+
+        /// <summary>
+        /// Metadata of the dataset.
+        /// </summary>
+        private Metadata _metadata;
+
+        /// <summary>
+        /// Container for storing spacial coordinates for every loaded spectrum.
+        /// </summary>
+        private List<SpatialCoordinates> _spatialCoordinates;
+
+        /// <summary>
+        /// Array of m/z values for all the spectras.
+        /// </summary>
+        private double[] _mz;
+
+        /// <summary>
+        /// Container for storing intensity values for every loaded spectrum.
+        /// </summary>
+        private List<double[]> _intensity;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor with file initialization.
+        /// </summary>
+        /// <param name="textFilePath">Path to the text file.</param>
+        public BasicTextDataset(string textFilePath)
+        {
+            CreateFromFile(textFilePath);
+        }
+        /// <summary>
+        /// Constructor with raw value initialization.
+        /// </summary>
+        /// <param name="mz">Array of m/z values.</param>
+        /// <param name="data">Array of intensity values.</param>
+        public BasicTextDataset(double[] mz, double[,] data)
+        {
+            CreateFromRawData(mz, data);
+        }
+
+        #endregion
+
+        #region IDataset
+
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public Metadata Metadata
+        {
+            get { return _metadata; }
+            private set { _metadata = value; }
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public IEnumerable<SpatialCoordinates> SpacialCoordinates
+        {
+            get { return _spatialCoordinates; }
+            private set { _spatialCoordinates = value as List<SpatialCoordinates>; }
+        }
+        /// <summary>
+        /// Method for creating new dataset from text file, overwriting current data.
+        /// </summary>
+        /// <param name="filePath">Path to a text file.</param>
+        /// <exception cref="Exception">Thrown where there is a problem with file loading.</exception>
+        public void CreateFromFile(string filePath)
+        {
+            try
+            {
+                StreamReader sr = new StreamReader(filePath);
+                var metadata = sr.ReadLine(); // global metadata
+                var mzValues = sr.ReadLine()?.Split(null);
+                _mz = new double[mzValues.Length];
+                for (int i = 0; i < _mz.Length; i++)
+                    if (!double.TryParse(mzValues[i], NumberStyles.Any, CultureInfo.InvariantCulture, out _mz[i]))
+                        _mz[i] = double.NaN;
+            }
+            catch (Exception e)
+            {
+                throw new Exception("File " + filePath + " could not be read.", e);
```

Comment:
... and throw? [`IoException` or its even more specific subclass](https://msdn.microsoft.com/pl-pl/library/system.io.ioexception(v=vs.110).aspx)?